src/material-experimental/selection/selection-column.ts

```diff
@@ -0,0 +1,102 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {MatCellDef, MatColumnDef, MatHeaderCellDef, MatTable} from '@angular/material/table';
+import {
+  Component,
+  Input,
+  isDevMode,
+  OnDestroy,
+  OnInit,
+  Optional,
+  ViewChild,
+  ChangeDetectionStrategy,
+  ViewEncapsulation,
+} from '@angular/core';
+
+import {MatSelection} from './selection';
+
+/**
+ * Column that adds row selecting checkboxes and a select-all checkbox if `matSelectionMultiple` is
+ * `true`.
+ *
+ * Must be used within a parent `MatSelection` directive.
+ */
+@Component({
+  selector: 'mat-selection-column',
+  template: `
+    <ng-container matColumnDef>
+      <th matHeaderCell *matHeaderCellDef align="center">
```

Comment:
It should be `mat-header-cell` rather than `matHeaderCell` (same for `matCell` below). This is why the selection column doesn't look like the rest of the table